### ISSUE TITLE
[4.0.0.rc2] Schedule / Periodic Tasks

### ISSUE THREAD

**Issue by honux** [celery/celery#3260]
The following example (http://docs.celeryproject.org/en/master/userguide/periodic-tasks.html#beat-entries):

```python
from celery import Celery
from celery.schedules import crontab

app = Celery(broker='redis://127.0.0.1/1', backend='redis://127.0.0.1/2')

@app.on_after_configure.connect
def setup_periodic_tasks(sender, **kwargs):
    # Calls test('hello') every 10 seconds.
    sender.add_periodic_task(10.0, test.s('hello'), name='add every 10')

    # Calls test('world') every 30 seconds
    sender.add_periodic_task(30.0, test.s('world'), expires=10)

    # Executes every Monday morning at 7:30 A.M
    sender.add_periodic_task(
        crontab(hour=7, minute=30, day_of_week=1),
        test.s('Happy Mondays!'),
    )

@app.task
def test(arg):
    print(arg)
```

Does not run the periodic tasks. The task run successfully if you call it using 'send_task'. It does not run even if you modify it directly:

```python
app.conf.beat_schedule = {
    'add-every-30-seconds': {
        'task': 'test.celery.test',
        'schedule': 30.0,
        'args': ('foo',)
    },
}
```

Trying to run with the latest celery, on Ubuntu 16 and Python 3

**Comment by ask** [comment]
Did you not get an exception?  For me it crashed, but maybe you were not looking at the beat logs?

**Comment by ask** [comment]
The fix in 7caafb4 likely solves it, since that would stop this example from working anywhere and it's probably been in master for some time.

If it does not solve it then we can reopen :)

**Comment by honux** [comment]
Yeah, there was a crash.
Now I'm getting this one, using the latest version:

```python
[2016-06-23 11:50:28,395: WARNING/Beat] DB Reset: Account for new __version__ field
[2016-06-23 11:50:28,395: ERROR/Beat] Process Beat
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/kombu/utils/__init__.py", line 300, in __get__
    return obj.__dict__[self.__name__]
KeyError: 'scheduler'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/billiard/process.py", line 303, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 602, in run
    self.service.start(embedded_process=True)
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 520, in start
    humanize_seconds(self.scheduler.max_interval))
  File "/usr/local/lib/python3.4/dist-packages/kombu/utils/__init__.py", line 302, in __get__
    value = obj.__dict__[self.__name__] = self.__get(obj)
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 561, in scheduler
    return self.get_scheduler()
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 556, in get_scheduler
    lazy=lazy)
  File "/usr/local/lib/python3.4/dist-packages/celery/utils/imports.py", line 53, in instantiate
    return symbol_by_name(name)(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 408, in __init__
    Scheduler.__init__(self, *args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 208, in __init__
    self.setup_schedule()
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 471, in setup_schedule
    self.merge_inplace(self.app.conf.beat_schedule)
  File "/usr/local/lib/python3.4/dist-packages/celery/beat.py", line 363, in merge_inplace
    entry = self.Entry(**dict(b[key], name=key, app=self.app))
TypeError: __init__() got an unexpected keyword argument 'route'
[2016-06-23 11:50:29,479: WARNING/MainProcess] Professional@ns522543.ip-158-69-224.net ready.
```

using:
```
@celery.on_after_configure.connect
def setup_periodic_tasks(sender, **kwargs):
    # Request update of the current match status every 60 seconds
    for region in ['br', 'eune', 'euw', 'jp', 'kr', 'lan', 'las', 'na', 'oce', 'tr', 'ru']:
        sender.add_periodic_task(60.0, signature('Professional.update_playing_professionals', args=(region,)))

```
